src/languageSupport/languages/flux/lsp/connection.ts

```diff
@@ -256,31 +170,32 @@ class LspConnectionManager {
     return [startLineStyle, middleLinesStyle, endLineStyle]
   }
 
-  _setEditorBlockStyle(schema: CompositionSelection = this._session) {
-    const compositionBlock = this._getCompositionBlockLines(
-      this._model.getValue()
-    )
-
-    const removeAllStyles = !compositionBlock || schema.composition.diverged
+  _setEditorBlockStyle(range: LspRange | null) {
+    const removeAllStyles = range == null
 
     this._compositionStyle = this._editor.deltaDecorations(
       this._compositionStyle,
       removeAllStyles
         ? []
-        : this._compositionSyncStyle(
-            compositionBlock?.startLine,
-            compositionBlock?.endLine,
-            schema.composition.synced
-          )
+        : this._compositionSyncStyle(range?.start.line, range.end.line)
```

Comment:
Just curious why `range?.start` but `range.end` without a `?`

My understanding is that line 174 already checked if range is null, then line 179 `[]` will be used, so it's good to use `range.start.line` without the question mark

Reply:
Thanks for catching, fixed!